components/Container/LayoutWithPanel.tsx

```diff
@@ -0,0 +1,237 @@
+'use client';
+
+import React, {
+  CSSProperties,
+  useState,
+  useEffect,
+  useRef,
+} from 'react';
+
+import { Button } from 'react-aria-components';
+
+import { DmpIcon } from '@/components/Icons';
+import {
+  useResponsive,
+  getSizeByName,
+} from '@/hooks/responsive';
+
+import { LayoutContainerProps } from '@/components/Container/LayoutContainer';
+import { ContentContainer } from '@/components/Container/ContentContainer';
+
+
+type DirectionType =
+  | null
+  | "left"
+  | "right";
+
+
+/**
+ * Layout with a dynamic sidebar
+ */
+interface LayoutWithPanelProps extends LayoutContainerProps {
+}
+
+export const LayoutWithPanel: React.FC<LayoutWithPanelProps> = ({
+  children,
+  id,
+  className = "",
+}) => {
+  const thisRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (thisRef) {
+      if (thisRef.current) {
+        let direction: DirectionType = null;
+
+        function _updateClassList(name: string) {
+          switch (name) {
+            case "ToolbarContainer": {
+              thisRef.current?.classList.add('with-toolbar');
+              break;
+            }
+
+            case "ContentContainer": {
+              if (!direction) direction = "right";
+              break;
+            }
+
+            case "SidebarPanel": {
+              thisRef.current?.classList.add('with-sidebar');
+              if (!direction) direction = "left";
+              break;
+            }
+
+            case "DrawerPanel": {
+              thisRef.current?.classList.add('with-drawer');
+              if (!direction) direction = "left";
+              break;
+            }
+          }
+        }
+
+        if (Array.isArray(children)) {
+          children?.forEach((rNode) => {
+            if (!rNode) return;
+            if (!React.isValidElement(rNode)) return;
+            _updateClassList(rNode.type.name);
```

Comment:
L76: @andrewebdev I'm seeing errors here because `rNode.type` might be a string, and therefore, there would be no `name` field on a string.

I believe you have to check whether rNode.type is a function or an object.